File: ios/Classes/Cause.swift
//
//  Cause.swift
//  zsdk
//
//  Created by Luis on 1/9/20.
//

import UIKit

enum Cause: String, CaseIterable {
    case PARTIAL_FORMAT_IN_PROGRESS = "PARTIAL_FORMAT_IN_PROGRESS"
    case HEAD_COLD = "HEAD_COLD"
    case HEAD_OPEN = "HEAD_OPEN"
    case HEAD_TOOHOT = "HEAD_TOOHOT"
    case PAPER_OUT = "PAPER_OUT"
    case RIBBON_OUT = "RIBBON_OUT"
    case RECEIVE_BUFFER_FULL = "RECEIVE_BUFFER_FULL"
    case UNKNOWN = "UNKNOWN"
}

File: lib/src/enumerators/cause.dart
/// Created by luis901101 on 2020-01-07.
enum Cause {
  PARTIAL_FORMAT_IN_PROGRESS,
  HEAD_COLD,
  HEAD_OPEN,
  HEAD_TOO_HOT,
  PAPER_OUT,
  RIBBON_OUT,
  RECEIVE_BUFFER_FULL,
  NO_CONNECTION,
  UNKNOWN,
}

extension CauseUtils on Cause {
  String get name => toString().split('.').last;

  static Cause? valueOf(String name) {
    for (Cause value in Cause.values) if (value.name == name) return value;
    return Cause.UNKNOWN;
  }
}
